scripts/cmake/xtensa-toolchain.cmake

```diff
@@ -45,3 +45,21 @@ set(CMAKE_FIND_ROOT_PATH  ".")
 set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
 set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
 set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
+
+if(XCC)
+	# get compiler description
+	execute_process(
+		COMMAND ${CMAKE_C_COMPILER} --show-config=config
+		OUTPUT_VARIABLE cc_config_output
+		OUTPUT_STRIP_TRAILING_WHITESPACE
+		ERROR_QUIET
+	)
+
+	string(REGEX MATCH "[a-zA-Z]+-[0-9]+.[0-9]+-[a-zA-Z]*" XCC_TOOLS_VERSION "${cc_config_output}")
+	if(NOT XCC_TOOLS_VERSION)
+		message(WARNING "Couldn't get compiler description from '${cc_config_output}'")
+		set(XCC_TOOLS_VERSION "${CMAKE_SYSTEM_NAME}")
```

Comment:
Sorry for late complain about this one - but I don't see why would you use OS NAME here, for me it's weird to have os in tools version that are usually year of tool release. Maybe it's better to just have `UNKNOWN` here?

Reply:
added 'UNKNOWN-${CMAKE_SYSTEM_NAME}'